slang.h

```diff
@@ -3240,16 +3269,50 @@ namespace slang
     }
 }
 
-/**
+/** Get the (linked) program for a compile request.
+
+The linked program will include all of the global-scope modules for the
```

Comment:
Would be nice to have SlangCompileRequest actually be an interface, that these functions could just be inline calls into that, if you wanted a 'c-like interface' still. Doing so would clean up some example code and probly other places cos can just use ComPtr.

Reply:
Revamping the compile request part of the API is definitely a next step. The biggest sticking point (for me) is whether it should stick with the current stateful model of the `sp...` functions, or whether it should use some kind of `Desc` structure. I know the latter is what most clients of the API would prefer, but we have also found that we keep having to add more bells and whistles to compilation.